CPS-????/README.md

```diff
@@ -0,0 +1,242 @@
+---
+CPS: ?
+Title: Spending Script Redundant Executions
+Status: Open
+Category: Plutus
+Authors: 
+  - fallen-icarus <modern.daidalos@gmail.com>
+Proposed Solutions: []
+Discussions:
+  - https://github.com/cardano-foundation/CIPs/issues/417
+Created: 2022-12-20
+---
+
+## Abstract
+Spendings scripts are currently executed once for every UTxO being consumed from a script address. However, there are use cases where the validity of the transaction depends on the transaction as a whole and not on any individual UTxO. The eUTxO architecture allows a spending script to see the entire transaction context which means it is already possible to create scripts that validate based on the transaction as a whole. As of now, spending scripts that do this are forced to be executed once per UTxO even if these extra executions are completely redundant. Not only does this waste computational resources, but it can also result in the transaction fee growing quickly based on the number ot UTxOs being consumed from the relevant script address. There is a trick that can be used to minimize this limitation but it has its own drawbacks. At the end of this CPS is a proposed solution (with a possible low-level specification) that not only addresses these redundant executions but would also make it possible to spend script utxos that have missing/wrong datums.
+
+## Problem
+Consider the following transaction:
+``` Bash
+cardano-cli transaction build \
+  --tx-in c1d01ea50fd233f9fbaef3a295ba607a72c736e58c9c9df588abf4e5009ad4fe#0 \
+  --tx-in 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#2 \
+  --spending-tx-in-reference 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#0 \
+  --spending-plutus-script-v2 \
+  --spending-reference-tx-in-inline-datum-present \
+  --spending-reference-tx-in-redeemer-file $swapRedeemerFile \
+  --tx-in 766555130db8ff7b50fc548cbff3caa0d0557ce5af804da3b993cd090f1a8c3a#1 \
+  --spending-tx-in-reference 622034715b64318e9e2176b7ad9bb22c3432f360293e9258729ce23c1999b9d8#0 \
+  --spending-plutus-script-v2 \
+  --spending-reference-tx-in-inline-datum-present \
+  --spending-reference-tx-in-redeemer-file $swapRedeemerFile \
+  --tx-out "$(cat ../assets/wallets/01.addr) 2000000 lovelace + 300 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.54657374546f6b656e0a + 0 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.4f74686572546f6b656e0a" \
+  --tx-out "$(cat ${swapScriptAddrFile}) + 4000000 lovelace + 0 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.54657374546f6b656e0a + 250 c0f8644a01a6bf5db02f4afe30d604975e63dd274f1098a1738e561d.4f74686572546f6b656e0a" \
+  --tx-out-inline-datum-file $swapDatumFile \
+  --tx-in-collateral bc54229f0755611ba14a2679774a7c7d394b0a476e59c609035e06244e1572bb#0 \
+  --change-address $(cat ../assets/wallets/01.addr) \
+  --protocol-params-file "${tmpDir}protocol.json" \
+  --testnet-magic 1 \
+  --out-file "${tmpDir}tx.body"
+```
```

Comment:
I don't think that showing this particularly lengthy command excerpt really serves the purpose of highlighting the problem. I'd suggest to simply go with prose of a succinct diagram showing how multiple inputs locked by the same script results in the script being executed multiple times.